src/Connection.php

```diff
@@ -40,9 +40,12 @@ class Connection extends EventEmitter implements ConnectionInterface
     public $stream;
 
     private $input;
+    private $options;
 
-    public function __construct($resource, LoopInterface $loop)
+    public function __construct($resource, LoopInterface $loop, array $options = array())
     {
+        $this->options = $options;
```

Comment:
See below, I think we *may* remove this parameter if we transport this as part of the context of the resource.